lib/hypervisor/hv_kvm/__init__.py

```diff
@@ -1850,17 +1855,27 @@ def _ExecuteKVMRuntime(self, instance, kvm_runtime: KVMRuntime,
           vhostfd = ""
 
         if kvm_supports_netdev:
+          dev_ops = []
+
           # Non paravirtual NICs hvinfo is empty
           if "id" in nic.hvinfo:
             nic_val = _GenerateDeviceHVInfoStr(nic.hvinfo)
             netdev = nic.hvinfo["id"]
           else:
             nic_val = "%s" % nic_model
             netdev = "netdev%d" % nic_seq
+
           nic_val += (",netdev=%s,mac=%s%s" % (netdev, nic.mac, nic_extra))
           tap_val = ("type=tap,id=%s,%s%s%s" %
                      (netdev, tapfd, vhostfd, tap_extra))
-          kvm_cmd.extend(["-netdev", tap_val, "-device", nic_val])
+          dev_ops.append(nic_val)
+
+          # add bootindex property to the first nic if network boot is enabled
+          if boot_network:
```

Comment:
For completeness, please check if `kernel_path` is unset, else the network boot can be skipped.